sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/DefaultGrpcServiceBuilder.java

```diff
@@ -147,17 +144,10 @@ public GrpcService<ReqT, ResT> build() {
         managedChannelBuilder.intercept(MetadataUtils.newAttachHeadersInterceptor(metadata));
       }
 
-      if (trustedCertificatesPem != null) {
-        try {
-          ManagedChannelUtil.setClientKeysAndTrustedCertificatesPem(
-              managedChannelBuilder, privateKeyPem, certificatePem, trustedCertificatesPem);
-        } catch (SSLException e) {
-          throw new IllegalStateException(
-              "Could not set trusted certificates for gRPC TLS connection, are they valid "
-                  + "X.509 in PEM format?",
-              e);
-        }
-      }
+      tlsConfigHelper.configureWithKeyManager(
+          (tm, km) ->
+              ManagedChannelUtil.setClientKeysAndTrustedCertificatesPem(
+                  managedChannelBuilder, tm, km));
```

Comment:
Its strange that we for the other gRPC clients (otlp, jaeger) we don't bother trying to configure the managed channel with certificates. We take the stance that if you bring your own `ManagedChannel`, you're responsible for configuring it. 

In fact, this is the only usage of `ManagedChannelUtil.setClientKeysAndTrustedCertificatesPem` outside of tests. 

We should adopt the same policy here. Doing so allows us to move `ManagedChannelUtil.setClientKeysAndTrustedCertificatesPem` to test code, where its used by `ManagedChannelTelemetryExporterBuilder`. This allows us to simplify `tlsConfigHelper` as we can get rid of the `configureWithKeyManager` method.

Reply:
Agreed, this is interesting and I would love to simplify. I was absolutely bummed when I discovered this "other path" for configuring TLS related stuff, and unifying/simplifying would be great. 

I think that removing `configureWithKeyManager()` from the helper would then require the `DefaultGrpcServiceBuilder` to call `setClientKeysAndTrustedCertificatesPem()` with a key manager and trust manager that it has constructed itself. I'm not sure immediately how to do with without duplicating some of the guts of `TlsConfigHelper`.

Reply:
> I think that removing configureWithKeyManager() from the helper would then require the DefaultGrpcServiceBuilder to call setClientKeysAndTrustedCertificatesPem() with a key manager and trust manager that it has constructed itself.

I'm suggesting we drop any configuration of `ManagedChannel` in `DefaultGrpcServiceBuilder`, as we've done with the other grpc clients that allow you to "bring your own channel". If we do so, the call to `setClientKeysAndTrustedCertificatesPem` goes away completely. I'm happy to do this in a separate PR to avoid expanding the scope of this PR.

Reply:
I support a separate PR (which I could also do if you want), since I think it will be a user-facing change.

Reply:
Go for it!

Reply:
Can we merge this first? There are a lot of cherries on this tree now....

Reply:
Sounds good to me